services/cd-service/pkg/repository/repository.go

```diff
@@ -208,7 +224,12 @@ func New(ctx context.Context, cfg Config) (Repository, error) {
 				}
 			}
 			// check that we can build the current state
-			if _, err := result.buildState(); err != nil {
+			state, err := result.buildState();
+			if err != nil {
+				return nil, err
+			}
+			
+			if err := state.validateConfigs(); err != nil {
```

Comment:
you could just use `state` and check whether `GetEnvironmentConfigs` returns an error. Then you would also check that the configs are semantically valid.

Reply:
Had originally discussed this with @sven-urbanski-freiheit-com , and he mentioned to keep validity check without semantics which is why the valid function was added

Reply:
`semantically valid` is a bit of a strong word for the checks we do there ;) But yes, we check a bit more.

But yes, I would agree with Hannes, you can basically just call `GetEnvironmentConfigs` and get a better result.
Just make sure that the error returned is actually readible (but that should already be the case) @mnishamk-freiheit

Reply:
,"msg":"repository.new.error","error":"invalid character ':' after object key:value pair",

this is the error if you use GetEnvironmnetConfigs, it comes from json.decoder.decode method. Might want to wrap this error in something readable pointing to the actual file